webapp/src/services/api.ts

```diff
@@ -224,15 +216,15 @@ export const api = createApi({
         "UtteranceCountPerFilter",
         "Utterances",
       ],
-      async onQueryStarted(
-        { persistentIds, newValue, ...args },
-        { dispatch, queryFulfilled }
-      ) {
+      async onQueryStarted({ body, ...args }, { dispatch, queryFulfilled }) {
         const patchResult = dispatch(
           api.util.updateQueryData("getUtterances", args, (draft) => {
             draft.utterances.forEach((utterance) => {
-              if (persistentIds.includes(utterance.persistentId)) {
-                utterance.dataAction = newValue;
+              const patchUtterance = body.find(
+                ({ persistentId }) => persistentId === utterance.persistentId
+              );
+              if (patchUtterance) {
+                utterance.dataAction = patchUtterance.dataAction;
```

Comment:
The name `patchUtterance` sounds like a verb and hints at a function. The name `utterancePatch`, matching the type of `body` would have been better. However, I think `found = body.find(...)` sounds even better, especially with `if (found)`. Do you like that?
```suggestion
              const found = body.find(
                ({ persistentId }) => persistentId === utterance.persistentId
              );
              if (found) {
                utterance.dataAction = found.dataAction;
```

Reply:
Sounds good to me. how about foundPersistenId or foundId?

Reply:
It would be a `foundUtterancePatch`, but I don't think this extra information is necessary, given the scope of this variable is so small.